contracts/B4REAL.sol

```diff
@@ -117,26 +131,46 @@ contract B4REAL is ERC20, AccessControl {
     {
         require(amount > 0, "The amount must be greater than 0");
 
-        uint256 tokensForTax;
-
         uint256 remainder = amount;
 
         // calculate the number of tokens the Tax should take
-        if (whitelist[to] && !waiveFees) {
-            tokensForTax = calculateFee(amount, taxFee, taxFeeDecimals);
-            remainder -= tokensForTax;
-        }
-        super.transfer(taxAddress, tokensForTax);
+        remainder = transferTax(msg.sender, to, amount);
+
         super.transfer(to, remainder);
         return true;
     }
 
-    function setAdmin(address admin) public onlyOwner {
-        grantRole(ADMIN_ROLE, admin);
+    function transferFrom(address sender, address recipient, uint256 amount)
+        public
+        override
+        returns (bool)
+    {
+        require(amount > 0, "The amount must be greater than 0");
+
+        uint256 remainder = amount;
+
+        // calculate the number of tokens the Tax should take
+        remainder = transferTax(sender, recipient, amount);
```

Comment:
Not sure what we're doing here?  Why the asignment on LN150 https://github.com/dltxio/b4real-contracts/pull/2/files#diff-e436e94acae1c0f6ce625442eb79d2f19efc384cb235e97988b53d6553a2ca73R150